.github/workflows/style.yaml

```diff
@@ -2,65 +2,84 @@ name: style
-        pip cache purge
-    - name: List pip dependencies
-      run: pip list
-    - name: Run mypy checks
-      run: mypy .
+      - name: Clone repo
+        uses: actions/checkout@v4.1.2
+      - name: Set up python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.12"
+      - name: Cache dependencies
+        uses: actions/cache@v4.0.2
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/required.txt') }}-${{ hashFiles('requirements/datasets.txt') }}-${{ hashFiles('requirements/style.txt') }}-${{ hashFiles('requirements/tests.txt') }}
+      - name: Install pip dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          pip install -r requirements/required.txt -r requirements/datasets.txt -r requirements/style.txt -r requirements/tests.txt
+          pip cache purge
+      - name: List pip dependencies
+        run: pip list
+      - name: Run mypy checks
+        run: mypy .
   ruff:
     name: ruff
     runs-on: ubuntu-latest
     steps:
-    - name: Clone repo
-      uses: actions/checkout@v4.1.3
-    - name: Set up python
-      uses: actions/setup-python@v5.1.0
-      with:
-        python-version: '3.12'
-    - name: Cache dependencies
-      uses: actions/cache@v4.0.2
-      id: cache
-      with:
-        path: ${{ env.pythonLocation }}
-        key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/style.txt') }}
-    - name: Install pip dependencies
-      if: steps.cache.outputs.cache-hit != 'true'
-      run: |
-        pip install -r requirements/style.txt
-        pip cache purge
-    - name: List pip dependencies
-      run: pip list
-    - name: Run ruff checks
-      run: |
-        ruff check --output-format=github --no-fix .
-        ruff format --diff
+      - name: Clone repo
+        uses: actions/checkout@v4.1.2
+      - name: Set up python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.12"
+      - name: Cache dependencies
+        uses: actions/cache@v4.0.2
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('requirements/style.txt') }}
+      - name: Install pip dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          pip install -r requirements/style.txt
+          pip cache purge
+      - name: List pip dependencies
+        run: pip list
+      - name: Run ruff checks
+        run: |
+          ruff check --output-format=github --no-fix .
+          ruff format --diff
+  prettier:
+    name: prettier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v4.1.2
+      - name: Set up nodejs
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+          cache: "npm"
```

Comment:
Does this actually help? I noticed that cache for setup-python didn't make installation any faster.

Reply:
Since it's already quite short I doubt that it will give any significant speedup, but haven't testes it either way, I just did noticed that you have used cache for python so I thought that on this one you will also wanna have it.

Some change will make using `npm ci` with cache instead of `npm install --save-dev --save-exact prettier`, in `npm` documentation it suggested that it have cut time by half but it was case where 150 dependencies have been installed.